### PR TITLE
[Lua] Implement remaining Geomantic Reservoirs

### PR DESCRIPTION
**Pull request by CatsEyeXI**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
Implement remaining Geomantic Reservoirs.

ID's used:
https://github.com/zach2good/UpdateExtractor/tree/versions/2022-june/out/scripts/zones

## Steps to test these changes

Starting with a fresh character, add Indi-Spells, then visit corresponding Geomantic reservoir to unlock Geo-Spell.

### CODE REVIEW

File: scripts/zones/Batallia_Downs/npcs/Geomantic_Reservoir.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
+-----------------------------------
+-- Area: Batallia Downs
+-- NPC: Geomantic Reservoir
+-- Unlocks: Geo-Barrier
+-- !pos -677.645 -32.000 157.981
```
Comment (zach2good) [inline scripts/zones/Batallia_Downs/npcs/Geomantic_Reservoir.lua]: All of these pos comments also need the zone number at the end so they can be traveled to from anywhere

File: scripts/zones/Beadeaux/IDs.lua [hunk 2]
```diff
@@ -32,6 +32,8 @@ zones[xi.zone.BEADEAUX] =
         FOUL_SMELLING_SOIL_IS_SPILLING_OUT = 7358, -- Foul-smelling soil is spilling out.
         CHEST_UNLOCKED                     = 7368, -- You unlock the chest!
         COMMON_SENSE_SURVIVAL              = 7795, -- It appears that you have arrived at a new survival guide provided by the Adventurers' Mutual Aid Network. Common sense dictates that you should now be able to teleport here from similar tomes throughout the world.
+        LEARNS_SPELL                       = 7786, -- <name> learns <spell>!
```
Comment (zach2good) [inline scripts/zones/Beadeaux/IDs.lua:34]: These are out if order

**Review by zach2good — CHANGES_REQUESTED** [review]
Almost there 👍

File: scripts/zones/Sauromugue_Champaign/IDs.lua [hunk 3]
```diff
@@ -45,8 +45,9 @@ zones[xi.zone.SAUROMUGUE_CHAMPAIGN] =
         VOIDWALKER_BREAK_KI      = 11005, -- The <keyitem> shatters into tiny fragments.
         VOIDWALKER_OBTAIN_KI     = 11006, -- Obtained key item: <keyitem>!
         COMMON_SENSE_SURVIVAL    = 12513, -- It appears that you have arrived at a new survival guide provided by the Adventurers' Mutual Aid Network. Common sense dictates that you should now be able to teleport here from similar tomes throughout the world.
+        LEARNS_SPELL             = 12504, -- <name> learns <spell>!
```
Comment (Omnione) [inline scripts/zones/Sauromugue_Champaign/IDs.lua:47]: These two need to be moved up to be in order

**Review by Omnione — CHANGES_REQUESTED** [review]
Other than the order, of a couple of text ID's, it looks good

File: scripts/zones/La_Theine_Plateau/npcs/Geomantic_Reservoir_2.lua [hunk 4]
```diff
@@ -0,0 +1,26 @@
+-----------------------------------
+-- Area: La Theine Plateau
+-- NPC: Geomantic Reservoir
+-- Unlocks: Geo-Regen
```
Comment (Omnione) [inline scripts/zones/La_Theine_Plateau/npcs/Geomantic_Reservoir_2.lua:4]: found a copy pasta, should be Geo-Refesh not Regen

**Review by Omnione — CHANGES_REQUESTED** [review]
Once that comment in the header for Geo-Refresh is changed this will be good to go

File: scripts/zones/La_Theine_Plateau/npcs/Geomantic_Reservoir_2.lua (hunk 4 above)
Comment (CatsEyeXI) [reply scripts/zones/La_Theine_Plateau/npcs/Geomantic_Reservoir_2.lua:4]: doh

**Review by Omnione — APPROVED** [review]

**Review by Xaver-DaRed — APPROVED** [review]

**Review by zach2good — APPROVED** [review]

**Pull request merged**